ui/src/pages/backoffice/Acquisition/Order/OrderDetails/OrderDetails.js

```diff
@@ -208,16 +207,13 @@ export default class OrderDetails extends React.Component {
   }
 
   componentDidMount() {
-    this.unlisten = history.listen(loc => {
-      if (loc.state && loc.state.pid && loc.state.type === 'Order') {
-        this.props.fetchOrderDetails(loc.state.pid);
-      }
-    });
     this.props.fetchOrderDetails(this.props.match.params.orderPid);
   }
 
-  componentWillUnmount() {
-    this.unlisten();
+  componentDidUpdate(prevProps) {
+    if (prevProps.match.params.orderPid !== this.props.match.params.orderPid) {
```

Comment:
General very minor comment (don't need to change everything!) for all these changes, to take into account in future code.
To keep the code more readable given the Prettier reformatting, I would have done for all of them:
```
const orderPid = this.props.match.params.orderPid
const samePidFromRouter = prevProps.match.params.orderPid === orderPid
if (!samePidFromRouter) {
    this.props.fetchOrderDetails(orderPid)
}
```
This is to remember that is coming from the Router and why we have that.
@KonstantinaStoikou WDYT?

Reply:
I agree with what you propose. Do you want me to change it?